### DISCUSSION TITLE
What is the difference between Argo CD and Argo Workflows?

### DISCUSSION THREAD

**Discussion by itssamuelrowe** [argoproj/argo-workflows#7435] [Q&A] [answered]
Hey. I've been reading about Argo CD. At our company, we considering Argo CD over Spinnaker. We need to implement pipelines. Eventually, I came across Argo Workflows. I'm a bit confused about what the difference between Argo CD and Argo Workflows is. It appears they have different dashboards and CLI tools. Can some please tell me what I am missing?

**Comment by terrytangyuan** [comment, marked as answer]
Argo Workflows is a generic framework for constructing pipelines/workflows and handles the orchestration and scheduling. There are many different use cases and some organizations use it for CI/CD.

Argo CD focuses on CD and is more for declarative and GitOps style continuous delivery on Kubernetes and you can use tools like kustomize/helm to manage manifests.

You can find some useful resources that help you understand the differences and use cases here: https://github.com/terrytangyuan/awesome-argo

**Reply by itssamuelrowe** [reply]
Argo Workflow, Argo CD, Argo Rollouts, and Argo Events... Are all of them independent projects decoupled from each other? What I mean is, I can use any of these tools individually. Is that right?

**Reply by terrytangyuan** [reply]
Yes

**Reply by marcellodesales** [reply]
Do we install all of them? ArgoCD first, then Workflows? what's their relationship?

**Reply by terrytangyuan** [reply]
You can use them independently or together.

**Reply by marcellodesales** [reply]
@terrytangyuan is there a possibility to have them installed together? Share the same UI? I wouldn't want to maintain 2 separate hosts since we are building a platform on top of CD and now Workflows... If there are instructions, could you please directions?

**Reply by terrytangyuan** [reply]
Currently, the UIs are separate. You can install Argo CD first and then use Argo CD to manage Argo Workflows as an application. When you are on Argo CD UI, you can click the Argo Workflows application's link to visit the Argo Workflows UI.

**Reply by rlabrecque** [reply]
Are there any best practices for using them together in a typical k8s container/microservice workflow? (post-install)

My thinking is what you end up doing is having an [argoproj-deployments](https://github.com/argoproj/argoproj-deployments) style repo with the deployment templates (as per the Argo CD best practices), then when you push to your applications code changes, argo-workflow builds that, pushes to your container registry, then updates the deployment registry with the new container tag, and then nudges argo-cd to refresh (to avoid waiting for the automatic resync?).

Is there anything else to it? How does one handle application changes which depend on configuration changes (Say a new env var)? Do you always have to commit config changes before application changes?

**Reply by airnorr** [reply]
...can I further ask, are there not elements of automated workflows and event triggers embedded in "ArgoCD" specific for continuous deployments? 

Would we need Argo events and workflows that can alert on changes to the repository?

**Comment by Zhang21** [comment]
tks!
